edits to demo 0, durable ID, based on phone calls this morning (#39)
microsoft/DynamicTelemetry: docs/docs/Demos.0.DynamicID.md

## Using DurableID in OpenTelemetry, to locate the source of the the expense

This demonstration will showcase and contrast two logging messages in
OpenTelemetry. One will use a Durable Identifier, and the other will not. It's
expected that not all software using OpenTelemetry will utilize a durable
identifier, even though, in Dynamic Telemetry, the use of the DurableID is
considered a best practice. A durable identifier functions like a GPS or homing
beacon, mapping any row of telemetry to a specific line of code.

> In Microsoft .net platform, a durable identifier is created automatically at
> compile time for OpenTelemetry user using ILogger - when the author of the
> software follows best practices.

Reviewer: ```suggestion
In Microsoft .NET platform, a durable identifier is created automatically at
compile time for OpenTelemetry user using `ILogger`, when the author of the
software follows best practices.
```

### Contrasting C# that makes use of Durable ID's, with C# that does not

Let's examine code that makes use of a durable identifier versus one that
doesn't. We'll inspect both in the Application Insights Log database, to spot
the difference.

> ### Seeing the DurableID in Azures Application Insights

Reviewer: ```suggestion
### Seeing the DurableID in Azure's Application Insights
```

> Lets start with the recommended way to log in .NET; using ILogger, with a
> compiler generated Durable ID.

Reviewer: ```suggestion
Let's start with the recommended way to log in .NET, using `ILogger` with [compile-time logging source generation] (https://learn.microsoft.com/en-us/dotnet/core/extensions/logger-message-generator).
```

#### Spotting the difference in your database

While both of these logging methods result in the same printf() ('flatted') log,

> the one with the Durable ID offers many extra goodies.

Reviewer: Formatting: The Durable Identifier/ durable identifier / Durable ID - it'd be nice to choose a single term and use it consistently across the doc.

You can see this quickly, with a quick dcount (distinct count) of each message.

![](../orig_media/Demo.0.DurableID.ShowCollisions.png)

Notice in this screenshot, that the number of distinct messages is similar to
the number of absolute messages - without a Durable Identifier, spotting the
differences can be difficult.  This in turn complicates cost reductions, as well
as any number of the Dynamic Telemetry scenarios.

## Use Azure Monitor and KQL language to locate an expensive OpenTelemetry log

Lets go back to the original code;  and use the compiler generated Durable ID to
quickly count the ILogger log, that followed recommended guidance.

```cdocs_include
{{ CSharp_Include("../Samples/DurableIds/Pages/Index.cshtml.cs",
    "// START : LoopRandomGUID",
    "// END : LoopRandomGUID")
}}
```

```cdocs_include
{{ CSharp_Include("../Samples/DurableIds/Pages/Index.cshtml.cs",
    "// START_DEFINE : LogRandomGuid",
    "// END_DEFINE : LogRandomGuid")
}}
```

This time, instead of trying to deduplicate with the message (where the GUID all
bug guarantees each row is unique), we will deduplicate with the
extracted EventName field.

> ![](../orig_media/Demo.0.DurableID.GroupedByEventName.png){width="5.5in"

Reviewer: Missed this one :D
```suggestion
![](../orig_media/Demo.0.DurableID.GroupedByEventName.png)
```

> ![](../orig_media/Demo.0.DurableID.GroupedByEventName.png){width="5.5in"
> height="4.3991393263342085in"}
> 

Reviewer: Missed this one :D
```suggestion
![](../orig_media/Demo.0.DurableID.GroupedByEventName.png)

```
